plugins/woocommerce/tests/api-core-tests/tests/customers/customers-crud.test.js

```diff
@@ -0,0 +1,451 @@
+			expect(responseJSON.message).toEqual('Invalid resource ID.');
+		});
+
+		test('can retrieve customers', async ({
+			request
+		}) => {
+			// call API to retrieve all customers should initially return empty array
+			const response = await request.get('/wp-json/wc/v3/customers');
+			const responseJSON = await response.json();
+			expect(response.status()).toEqual(200);
+			expect(Array.isArray(responseJSON));
+			expect(responseJSON.length).toEqual(0);
+		});
+
+		// however, if we pass in the search string for role 'all' then all users are returned
+		test('can retrieve all customers', async ({
+			request
+		}) => {
+			// call API to retrieve all customers should initially return empty array
+            // unless the role 'all' is passed as a search string, in which case the admin
+            // and subscriber users will be returned
+			const response = await request.get('/wp-json/wc/v3/customers', {
+				params: {
+					role: 'all'
+				}
+			});
+			const responseJSON = await response.json();
+			expect(response.status()).toEqual(200);
+			expect(Array.isArray(responseJSON));
+			expect(responseJSON.length).toBeGreaterThanOrEqual(2);
+		});
+	});
+
+	test.describe('Create a customer', () => {
+		test('can create a customer', async ({
+			request,
+		}) => {
+			// call API to create a customer
+			const response = await request.post('/wp-json/wc/v3/customers', {
+				data: customer,
+			});
+			const responseJSON = await response.json();
+
+			// Save the customer ID. It will be used by the retrieve, update, and delete tests.
+			customerId = responseJSON.id;
+
+			expect(response.status()).toEqual(201);
+			expect(typeof responseJSON.id).toEqual('number');
+			// Verify that the customer role is 'customer'
+			expect(responseJSON.role).toEqual('customer');
+		});
+	});
+
+	test.describe('Retrieve after create', () => {
+		test('can retrieve a customer', async ({
+			request
+		}) => {
+			// call API to retrieve the previously saved customer
+			const response = await request.get(
+				`/wp-json/wc/v3/customers/${customerId}`
+			);
+			const responseJSON = await response.json();
+			expect(response.status()).toEqual(200);
+			expect(responseJSON.id).toEqual(customerId);
+			expect(responseJSON.is_paying_customer).toEqual(false);
+			expect(responseJSON.role).toEqual('customer');
+		});
+
+		test('can retrieve all customers', async ({
+			request
+		}) => {
+			// call API to retrieve all customers
+			const response = await request.get('/wp-json/wc/v3/customers');
+			const responseJSON = await response.json();
+			expect(response.status()).toEqual(200);
+			expect(Array.isArray(responseJSON));
+			expect(responseJSON.length).toBeGreaterThan(0);
+		});
+	});
+
+	test.describe('Update a customer', () => {
+
+		test(`can update the admin user/customer`, async ({
+			request,
+		}) => {
+			// update customer names (regular, billing and shipping) to admin
+			// (these were initialised blank)
```

Comment:
Mint be worth mentioning where that initialization happened so it's clear whether this is an issue/bug.

Reply:
Good shout - added link to the to the [GH documentation](https://github.com/woocommerce/woocommerce/tree/trunk/plugins/woocommerce/tests/e2e-pw#woocommerce-playwright-end-to-end-tests) for the environment